process/block/shardblock.go

```diff
@@ -516,6 +516,13 @@ func (sp *shardProcessor) indexBlockIfNeeded(
 		return
 	}
 
+	headerHash, err := core.CalculateHash(sp.marshalizer, sp.hasher, header)
```

Comment:
You already has headerHash computed in CommitBlock method from where you call this method. You can send this parameter and avoid to recalculate it again here

Reply:
done